File: src/main/java/me/iankit/simplechat/chat/GroupService.java
package me.iankit.simplechat.chat;

import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Service
public class GroupService {
    private final GroupDao groupDao;

    public GroupService(GroupDao groupDao) {
        this.groupDao = groupDao;
    }

    public Mono<Group> createGroup(CreateGroupDto dto) {
        return groupDao.save(dto.toEntity());
    }

    public Flux<Group> getAllGroups() {
        return groupDao.findAll();
    }
}

File: src/main/java/me/iankit/simplechat/chat/GroupService.kt
package me.iankit.simplechat.chat

import org.springframework.stereotype.Service
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@Service
class GroupService(private val groupDao: GroupDao) {
    fun createGroup(dto: CreateGroupDto): Mono<Group> {
        return groupDao.save(dto.toEntity())
    }

    fun getGroupById(groupId: String): Mono<Group> {
        return groupDao.findById(groupId)
    }

    fun allGroups(): Flux<Group> {
        return groupDao.findAll()
    }
}